docs/pre-release/reference/routing.md

```diff
@@ -37,6 +45,19 @@ curl some-host
 ```
 results in a http request with header `Host: some-host`. Now, if a service-mesh like Istio performs header based routing, then it will fail to find that host unless the request originates from the same namespace as the host resides in. Another reason is that the configuration of a service mesh can contain very strict rules. If the request then originates from the wrong pod, it will be denied. Only one intercept at a time can be used if there is a need to ensure that the chosen pod is exactly right.
 
+### Recursion detection
+It is common that clusters used in development, such as Minikube, Minishift or k3s, runs on the same host as the Telepresence client, often in a Docker container. Such clusters may have access to host network which means that both DNS and L4 routing may be subjected to recursion.
+
+#### DNS recursion
+When a local cluster's DNS-resolver fails to resolve a hostname, it may fall back to querying the local host network. This ultimately means that the Telepresence resolver will be asked to resolve a query that was issued from the cluster. Telepresence must check if such a query is recursive because there is a chance that it actually originated from the Telepresence DNS resolver and was dispatched to the `traffic-manager`, or a `traffic-agent`.
+
+Telepresence handles this by sending one initial DNS-query to resolve the hostname "tel2-recursion-check.kube-system". If the cluster runs locally, and have access to the local host's network, then that query will recurse back into the Telepresence resolver. Telepresence remembers this and alters its own behavior so that queries that are believed to be recursions are detected and respond with an NXNAME record. This is a best-effort type of solution. There's a chance that the DNS-resolver will yield a false negative for the second query if the same hostname is queried more than once in rapid succession, i.e. when the second query is made before the first query has received a response from the cluster.
```

Comment:
"and has access to[...]" not "and have access"
I'd rephrase "This is a best-effort type of solution" to "Telepresence performs this solution to the best of its ability, but may not be completely accurate in all situations.
Also, [avoid latinisms](https://www.notion.so/datawire/Latinisms-ba2ef26bca564c0f951f41dd66ca57ea). Swap "i.e." with "such as."